packages/website/ts/pages/cfl/cfl_metrics.tsx

```diff
@@ -0,0 +1,197 @@
+    }
+`;
+
+interface CFLMetricsProps {}
+
+interface CFLMetricsState {
+    cflMetricsData: CFLMetricsPairData[];
+    selectedIndex: number;
+}
+
+export class CFLMetrics extends React.Component<CFLMetricsProps, CFLMetricsState> {
+    public state: CFLMetricsState = {
+        selectedIndex: 0,
+        cflMetricsData: defaultData,
+    };
+    public componentDidMount(): void {
+        // tslint:disable-next-line:no-floating-promises
+        this._updateCFLMetricsDataAsync();
+    }
+    public render(): React.ReactNode {
+        const { cflMetricsData, selectedIndex } = this.state;
+        const quoteToken = this._getSelectedPairData().quoteAssetSymbol;
+        return (
+            <CFLMetricsContainer>
+                <PairTabsContainer>
+                    {cflMetricsData.map((data, index) => {
+                        const symbol = `${data.baseAssetSymbol} / ${data.quoteAssetSymbol}`;
+                        return (
+                            <PairTab
+                                key={symbol}
+                                isSelected={selectedIndex === index}
+                                onClick={this._onTabClick.bind(this, index)}
+                            >
+                                {symbol}
+                            </PairTab>
+                        );
+                    })}
+                </PairTabsContainer>
+                <MetricsContainer>
+                    <Metrics title={`Price in ${quoteToken}`} metrics={[{ value: this._getLastPrice() }]} />
+                    <Metrics title="7 day volume" metrics={[{ value: this._getVolume() }]} />
+                </MetricsContainer>
+                <MetricsContainer>
+                    <Metrics
+                        title={`7 day average slippage for $10,000`}
+                        info={SLIPPAGE_TOOLTIP_TEXT}
+                        metrics={this._getSlippageMetrics()}
+                    />
+                </MetricsContainer>
+                <FreshnessIndicator>
+                    <Paragraph size="small" textAlign="right">
+                        Updates every 30 minutes.
+                    </Paragraph>
+                </FreshnessIndicator>
+            </CFLMetricsContainer>
+        );
+    }
+    private async _updateCFLMetricsDataAsync(): Promise<void> {
+        try {
+            const data = await backendClient.getCFLMetricsAsync();
+            this.setState({
+                cflMetricsData: data,
+            });
+        } catch (err) {
+            // Do nothing. Will remain in loading state.
+        }
+    }
+    private _onTabClick(index: number): void {
+        this.setState({ selectedIndex: index });
+    }
+    private _getSelectedPairData(): CFLMetricsPairData {
+        return this.state.cflMetricsData[this.state.selectedIndex];
+    }
+    private _getLastPrice(): string {
+        const data = this._getSelectedPairData();
+        if (!data.lastTradePrice) {
+            return '—';
```

Comment:
should we give some indication of loading other than the dashes?

Reply:
nah